src/components/admin/SemesterEditDialog.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useContext } from "react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { useForm, type SubmitHandler } from "react-hook-form";
+import { z } from "zod";
+
+import { parseDate, validateDate } from "@/lib/utils";
+import { DialogCard, DialogCardFooter } from "../DialogUtils";
+import { OptionsDialogContext } from "../OptionsPopover";
+import { TextInput } from "../TextInput";
+import { useToast } from "../ui/use-toast";
+import { SemesterContext } from "./SemestersContext";
+
+const formSchema = z.object({
+  startDate: z
+    .string()
+    .min(1, "Field is required")
+    .refine(validateDate, "Invalid date"),
+  endDate: z
+    .string()
+    .min(1, "Field is required")
+    .refine(validateDate, "Invalid date"),
+  breakStart: z
+    .string()
+    .min(1, "Field is required")
+    .refine(validateDate, "Invalid date"),
+  breakEnd: z
+    .string()
+    .min(1, "Field is required")
+    .refine(validateDate, "Invalid date"),
+});
+
```

Comment:
Can we also validate that:

`startDate < breakStart`
`breakStart < breakEnd`
`breakEnd < endDate`

You can use zod's `.refine()` or `.superRefine()` for this.

https://zod.dev/?id=refine